src/components/permissionable/permissionable.cairo

```diff
@@ -1,3 +1,109 @@
+    use starknet::{ContractAddress, get_caller_address, get_block_timestamp};
+    use token_bound_accounts::components::account::account::AccountComponent;
+    use token_bound_accounts::interfaces::IAccount::{IAccount, IAccountDispatcherTrait};
+    use token_bound_accounts::components::account::account::AccountComponent::InternalImpl;
+    use token_bound_accounts::interfaces::IPermissionable::{
+        IPermissionable, IPermissionableDispatcher, IPermissionableDispatcherTrait
+    };
 
+    // *************************************************************************
+    //                              STORAGE
+    // *************************************************************************
+    #[storage]
+    pub struct Storage {
+        permissions: Map<
+            (ContractAddress, ContractAddress), bool
+        > // <<owner, permissioned_address>, bool>
+    }
 
+    // *************************************************************************
+    //                              EVENTS
+    // *************************************************************************
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        PermissionUpdated: PermissionUpdated
+    }
+
+    // @notice emitted when permissions are updated for an account
+    // @param owner tokenbound account owner
+    // @param permissioned_address address to be given/revoked permission
+    // @param has_permission returns true if user has permission else false
+    #[derive(Drop, starknet::Event)]
+    pub struct PermissionUpdated {
+        #[key]
+        pub owner: ContractAddress,
+        pub permissioned_address: ContractAddress,
+        pub has_permission: bool,
+    }
+
+    // *************************************************************************
+    //                              ERRORS
+    // *************************************************************************
+    pub mod Errors {
+        pub const INVALID_LENGTH: felt252 = 'Account: invalid length';
+    }
+
+
+    // *************************************************************************
+    //                              EXTERNAL FUNCTIONS
+    // *************************************************************************
+    #[embeddable_as(PermissionableImpl)]
+    pub impl Permissionable<
+        TContractState,
+        +HasComponent<TContractState>,
+        +Drop<TContractState>,
+        impl Account: AccountComponent::HasComponent<TContractState>
+    > of IPermissionable<ComponentState<TContractState>> {
+        // @notice sets permission for an account
+        // @permissioned_addresses array of addresses who's permission is to be updated
+        // @param permssions permission value <true, false>
+        fn set_permission(
+            ref self: ComponentState<TContractState>,
+            permissioned_addresses: Array<ContractAddress>,
+            permissions: Array<bool>
+        ) {
```

Comment:
Was going over our implementation again and it just came to mind that we should enforce only the contract owner calling this function. else permissioned addresses might give permissions to random addresses on the owner's account.

So let's enforce an assert that checks that the caller is the account owner.